### ISSUE TITLE
Bug: `eslint --init` and `eslint --mcp` swallow child process failures

### ISSUE THREAD

**Issue by fallintoplace** [eslint/eslint#20915]
### Environment

Node version: v24.15.0
npm version: 11.12.1
Local ESLint version: main @ 64bca24e7bed35bc3c864fc625cb2d89eca87d5b
Global ESLint version: N/A
Operating System: Darwin 25.5.0 arm64

### What parser are you using?

Default (Espree)

### What did you do?

I checked `bin/eslint.js` and then reproduced the behavior by forcing `cross-spawn` to report a child failure.

Both special-case branches call `spawn.sync(...)` and immediately `return`, without checking `result.status`, `result.signal`, or `result.error`.

Minimal shell repro from a checkout of `main`:

```sh
tmpdir=$(mktemp -d)
mkdir -p "$tmpdir/cross-spawn"
cat > "$tmpdir/cross-spawn/index.js" <<'EOF2'
module.exports = {
  sync() {
    return { status: 7 };
  }
};
EOF2

NODE_PATH="$tmpdir" node bin/eslint.js --mcp
printf 'exit=%s\n' $?

NODE_PATH="$tmpdir" node bin/eslint.js --init
printf 'exit=%s\n' $?
```

I also tried returning `{ error: new Error("boom") }` from the stub, and both commands still exited 0.

A bit of history in case it helps narrow this down:

- `--init` got this wrapper behavior in `0d2b9a6` when the command was moved to `npm init @eslint/config`
- `--mcp` got the same pattern in `ba456e0` when the server was externalized to `@eslint/mcp`
- the `--mcp` test update in `06a22f1` only checks startup output, not failure propagation

### What did you expect to happen?

If the delegated command exits nonzero, `eslint --init` / `eslint --mcp` should also exit nonzero.

If `spawn.sync()` returns an `error`, the wrapper should surface that failure instead of reporting success.

### What actually happened?

Both commands exited 0 in the repro above, even though the child result reported failure.

That means a failing `npm init @eslint/config@latest` or `npx @eslint/mcp@latest` can look successful to CI/scripts that only inspect the wrapper's exit status.

### Link to Minimal Reproducible Example

Minimal shell repro is included above.

### Additional comments

A fix here probably needs both behavior and test coverage:

- propagate nonzero child exit status
- surface `result.error`
- add tests for `--init` and `--mcp` failure propagation, not just successful startup

**Comment by DMartens** [comment]
Thank you for taking the time to file an issue.
I agree that in this case the error code and error of the spawned commands should be forwarded.
This seems to be oversight (I could not find anything in the relating PRs or RFCs).
As such marking as accepted.
